sys/include/net/netstats.h

```diff
@@ -48,6 +48,7 @@ typedef struct {
                                      (either acknowledged or unconfirmed
                                      sending operation, e.g. multicast) */
     uint32_t tx_failed;         /**< failed sending operations */
+    uint32_t tx_retrans;        /**< total number of retransmissions */
```

Comment:
AFAIK (at least on the Atmel radio) this is not the total number of retransmissions but the number of transmissions where all retransmission attemts failed to get an ACK back.

Reply:
I think that comment was supposed to the event handler? And isn't this the same? At least that is what we used in the past to count L2 retransmissions in our experiments. Or am I misunderstanding something?

Reply:
Ideally, we would count the absolute number of retransmission's here but this is not always the case (only speaking for the Atmel radio, we would need to lookup others). When this interrupt fires, MAX_RETIRES retransmissions failed to get an ACK, which is 4 by default. When an ACK appears after the second retransmission you won't notice.

Reply:
If this event imposes radio-specific corner cases to the upper layer, the design is already broken. So we should really be sure what this event does exactly.

Reply:
> So we should really be sure what this event does exactly.

Behavior of the Atmel radio was explained above. Next would be to find out handling on other devices.

Reply:
No. The event should be the same for every device. Otherwise the abstraction of netdev is broken.

Reply:
Neither you nor do I know if the event/metric is the same for every device (or not) so we can't answer the question unless you find out.

Reply:
Yes, we should look into this and fix it in case there are different behaviors. I will do so, when I find the time.

Reply:
I remember a really old question from @OlegHahm asking if I knew how to get the hardware auto-retransmission counter, and I think we did not find anything.
So if the re-transmission is done by the hardware itself, it would not be noticed.

I think that is what Peter is talking about.

Reply:
@PeterKietzmann I think in some old branch of mine I fiddled around with setting the retry amount to zero and handle the retransmissions within the driver, but I don't remember the details any more.

Reply:
Thanks for your feedback. @cladmi yes, at least this is the case for AT86RF2XX. No idea about other devices that we support. @OlegHahm yes I know, that was even a (never merged) PR of yours that I reference from time to time.
To conclude: If "all" other devices than the Atmel radio do support this count, we could go with this PR. If not we might delay the discussion until there is more from the software MAC initiative (by @jia200x )

Reply:
Ok. I'm leaving this PR open for now for later reference though (or if we need the counter for experiments again ;-)).

Reply:
It should not be used for experiments with `samr21-xpro` or `iotlab-m3` though or results would be based on a too low value.